vault/identity_store_oidc_provider.go

```diff
@@ -171,3 +222,108 @@ func (i *IdentityStore) pathOIDCAssignmentExistenceCheck(ctx context.Context, re
 
 	return entry != nil, nil
 }
+
+// pathOIDCCreateUpdateScope is used to create a new named scope or update an existing one
+func (i *IdentityStore) pathOIDCCreateUpdateScope(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	name := d.Get("name").(string)
+
+	var scope namedScope
+	if req.Operation == logical.UpdateOperation {
+		entry, err := req.Storage.Get(ctx, namedScopePath+name)
+		if err != nil {
+			return nil, err
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(&scope); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if templateRaw, ok := d.GetOk("template"); ok {
```

Comment:
We'll want to attempt to base64 decode and validate the template similar to how they're treated for identity token roles: [identity_store_oidc.go#L955-L984](https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc.go#L955-L984). There are also some reserved claims that we'll want to prevent the template from overwriting.